Add capacity to get yearly grouping for microposts.

ghlecl opened the pull request:
Adds a parameter called YearlyMicroposts which is used to either:
   - output the usual list of microposts if the parameter is not set or set to
     false;
   - output a list of micropost titles grouped by year (just like the posts list
     is) if the parameter is set and true.
The default behavior is not changed and the setting needs not be defined thus
not complicating the default installation.

I don't know if that is a modification that would be acceptable to you.  I am trying to keep my fork in sync with yours.  I have merged all your latest versions.  I have made sure that (as far as I can tell) there is no modification needed for the default behavior to remain.

It turns out that my microposts are not all that micro, they are more like mini posts.  Because of that, and because of how I tend to look for them, I would like to list them out just like my posts, but still keep them as a separate "type".  That is why I introduced this functionality.  I did not think a third kind of document (posts, miniposts and microposts or something like that) was the way to go.  I do not think it has much effect on sites, as I mentioned.  I have tested with and without the setting and with false and true as the values.

If you feel it's too much trouble and not worth it or something you are not happy with, I understand completely and I'll try and keep it in my fork and continue merging your modifications.

Thanks for the theme by the way.  Really happy with it.

ghlecl committed:
Add capacity to get yearly grouping for microposts.

Adds a parameter called YearlyMicroposts which is used to either:
   - output the usual list of microposts if the parameter is not set or set to
     false;
   - output a list of micropost titles grouped by year (just like the posts list
     is) if the parameter is set and true.
The default behavior is not changed and the setting needs not be defined thus
not complicating the default installation.

jnjosh commented:
@ghlecl Thanks for the pull request and thanks for using the theme. Sorry for the delay, I mistakenly turned off github notifications. 

I think this is fine but would you mind also updating the `README.md` to define the parameter and what to expect when setting it? I'd like to keep it well documented.

ghlecl commented:
@jnjosh  My pleasure.  Actually happy you are not against the modification.

I will update the README.md as soon as I get a chance (at work now) and I will get back to you.  I will also merge in your latest commit and see if it changes anything for me.  I don't think so at first glance.

I have never made a new commit after a pull request.  Will the pull request include the new commit?  Anyhow, I guess I'll learn.

Thanks again.

jnjosh commented:
@ghlecl I'm all for making this theme better 👍  

The pull request will include the new commit as long as you commit to the same branch. Pull requests are based on branches so it should _just work_.

ghlecl committed:
Add list class to default list as jnjosh did.

ghlecl committed:
Updated documentation to explain the new YearlyMicroposts parameter.

ghlecl commented:
@jnjosh I have added the list class in the default list as you did in your last commit and I have added some text to the README about the yearly microposts.  I have also corrected the flickrid text in the readme.  I had not realized that the pull requests are branch based last time and I had made a modification where now, if no flickr ID is provided, that section is simply removed from the footer.

Let me know what you think and if you want me to give more details or if it seems comprehensible.

jnjosh reviewed (approved):
Looks good! 👍
